components/calendar/src/iso.rs

```diff
@@ -135,7 +135,9 @@ impl Calendar for Iso {
 
         // The days of the week are the same every 400 years
         // so we normalize to the nearest multiple of 400
-        let years_since_400 = date.0.year % 400;
+        let years_since_400 = date.0.year.rem_euclid(400);
+        debug_assert!(years_since_400 >= 0); // rem_euclid returns positive numbers
```

Comment:
it does, let's not assert std guarantees.

Reply:
I suspected you might leave a comment like this :) ... I still have a slight preference for assertions on code invariants since `as` casts can be error-prone